// @ionic-native/background-mode/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Injectable } from '@angular/core';
import { Cordova, IonicNativePlugin, Plugin } from '@ionic-native/core';
import { Observable } from 'rxjs/Observable';
/**
 * @name Background Mode
 * @description
 * Cordova plugin to prevent the app from going to sleep while in background.
 * Requires Cordova plugin: cordova-plugin-background-mode. For more info about plugin, visit: https://github.com/katzer/cordova-plugin-background-mode
 * @usage
 * ```typescript
 * import { BackgroundMode } from '@ionic-native/background-mode';
 *
 * constructor(private backgroundMode: BackgroundMode) { }
 *
 * ...
 *
 * this.backgroundMode.enable();
 * ```
 *
 * @interfaces
 * BackgroundModeConfiguration
 */
var BackgroundMode = (function (_super) {
    __extends(BackgroundMode, _super);
    function BackgroundMode() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * Enable the background mode.
     * Once called, prevents the app from being paused while in background.
     */
    /**
       * Enable the background mode.
       * Once called, prevents the app from being paused while in background.
       */
    BackgroundMode.prototype.enable = /**
       * Enable the background mode.
       * Once called, prevents the app from being paused while in background.
       */
    function () { };
    /**
     * Disable the background mode.
     * Once the background mode has been disabled, the app will be paused when in background.
     */
    /**
       * Disable the background mode.
       * Once the background mode has been disabled, the app will be paused when in background.
       */
    BackgroundMode.prototype.disable = /**
       * Disable the background mode.
       * Once the background mode has been disabled, the app will be paused when in background.
       */
    function () {
        return;
    };
    /**
     * Checks if background mode is enabled or not.
     * @returns {boolean} returns a boolean that indicates if the background mode is enabled.
     */
    /**
       * Checks if background mode is enabled or not.
       * @returns {boolean} returns a boolean that indicates if the background mode is enabled.
       */
    BackgroundMode.prototype.isEnabled = /**
       * Checks if background mode is enabled or not.
       * @returns {boolean} returns a boolean that indicates if the background mode is enabled.
       */
    function () {
        return;
    };
    /**
     * Can be used to get the information if the background mode is active.
     * @returns {boolean} returns a boolean that indicates if the background mode is active.
     */
    /**
       * Can be used to get the information if the background mode is active.
       * @returns {boolean} returns a boolean that indicates if the background mode is active.
       */
    BackgroundMode.prototype.isActive = /**
       * Can be used to get the information if the background mode is active.
       * @returns {boolean} returns a boolean that indicates if the background mode is active.
       */
    function () {
        return;
    };
    /**
     * Override the default title, ticker and text.
     * Available only for Android platform.
     * @param {BackgroundModeConfiguration} options List of option to configure. See table below
     * @returns {Promise<any>}
     */
    /**
       * Override the default title, ticker and text.
       * Available only for Android platform.
       * @param {BackgroundModeConfiguration} options List of option to configure. See table below
       * @returns {Promise<any>}
       */
    BackgroundMode.prototype.setDefaults = /**
       * Override the default title, ticker and text.
       * Available only for Android platform.
       * @param {BackgroundModeConfiguration} options List of option to configure. See table below
       * @returns {Promise<any>}
       */
    function (options) {
        return;
    };
    /**
     * Modify the displayed information.
     * Available only for Android platform.
     * @param {BackgroundModeConfiguration} [options] Any options you want to update. See table below.
     */
    /**
       * Modify the displayed information.
       * Available only for Android platform.
       * @param {BackgroundModeConfiguration} [options] Any options you want to update. See table below.
       */
    BackgroundMode.prototype.configure = /**
       * Modify the displayed information.
       * Available only for Android platform.
       * @param {BackgroundModeConfiguration} [options] Any options you want to update. See table below.
       */
    function (options) { };
    /**
     * Listen for events that the plugin fires. Available events are `enable`, `disable`, `activate`, `deactivate` and `failure`.
     * @param event {string} Event name
     * @returns {Observable<any>}
     */
    /**
       * Listen for events that the plugin fires. Available events are `enable`, `disable`, `activate`, `deactivate` and `failure`.
       * @param event {string} Event name
       * @returns {Observable<any>}
       */
    BackgroundMode.prototype.on = /**
       * Listen for events that the plugin fires. Available events are `enable`, `disable`, `activate`, `deactivate` and `failure`.
       * @param event {string} Event name
       * @returns {Observable<any>}
       */
    function (event) {
        return;
    };
    /**
     * Android allows to programmatically move from foreground to background.
     */
    /**
       * Android allows to programmatically move from foreground to background.
       */
    BackgroundMode.prototype.moveToBackground = /**
       * Android allows to programmatically move from foreground to background.
       */
    function () { };
    /**
     * Enable GPS-tracking in background (Android).
     */
    /**
       * Enable GPS-tracking in background (Android).
       */
    BackgroundMode.prototype.disableWebViewOptimizations = /**
       * Enable GPS-tracking in background (Android).
       */
    function () { };
    /**
     * Android allows to programmatically move from background to foreground.
     */
    /**
       * Android allows to programmatically move from background to foreground.
       */
    BackgroundMode.prototype.moveToForeground = /**
       * Android allows to programmatically move from background to foreground.
       */
    function () { };
    /**
     * Override the back button on Android to go to background instead of closing the app.
     */
    /**
       * Override the back button on Android to go to background instead of closing the app.
       */
    BackgroundMode.prototype.overrideBackButton = /**
       * Override the back button on Android to go to background instead of closing the app.
       */
    function () { };
    /**
     * Exclude the app from the recent task list. Works on Android 5.0+.
     */
    /**
       * Exclude the app from the recent task list. Works on Android 5.0+.
       */
    BackgroundMode.prototype.excludeFromTaskList = /**
       * Exclude the app from the recent task list. Works on Android 5.0+.
       */
    function () { };
    /**
     * The method works async instead of isActive() or isEnabled().
     * @returns {Promise<boolean>}
     */
    /**
       * The method works async instead of isActive() or isEnabled().
       * @returns {Promise<boolean>}
       */
    BackgroundMode.prototype.isScreenOff = /**
       * The method works async instead of isActive() or isEnabled().
       * @returns {Promise<boolean>}
       */
    function () {
        return;
    };
    /**
     * Turn screen on
     */
    /**
       * Turn screen on
       */
    BackgroundMode.prototype.wakeUp = /**
       * Turn screen on
       */
    function () { };
    /**
     * Turn screen on and show app even locked
     */
    /**
       * Turn screen on and show app even locked
       */
    BackgroundMode.prototype.unlock = /**
       * Turn screen on and show app even locked
       */
    function () { };
    BackgroundMode.decorators = [
        { type: Injectable },
    ];
    __decorate([
        Cordova({
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "enable", null);
    __decorate([
        Cordova(),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], BackgroundMode.prototype, "disable", null);
    __decorate([
        Cordova({
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Boolean)
    ], BackgroundMode.prototype, "isEnabled", null);
    __decorate([
        Cordova({
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Boolean)
    ], BackgroundMode.prototype, "isActive", null);
    __decorate([
        Cordova({
            platforms: ['Android']
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", Promise)
    ], BackgroundMode.prototype, "setDefaults", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [Object]),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "configure", null);
    __decorate([
        Cordova({
            observable: true,
            clearFunction: 'un',
            clearWithArgs: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", [String]),
        __metadata("design:returntype", Observable)
    ], BackgroundMode.prototype, "on", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "moveToBackground", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "disableWebViewOptimizations", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "moveToForeground", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "overrideBackButton", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "excludeFromTaskList", null);
    __decorate([
        Cordova({
            platforms: ['Android']
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", Promise)
    ], BackgroundMode.prototype, "isScreenOff", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "wakeUp", null);
    __decorate([
        Cordova({
            platforms: ['Android'],
            sync: true
        }),
        __metadata("design:type", Function),
        __metadata("design:paramtypes", []),
        __metadata("design:returntype", void 0)
    ], BackgroundMode.prototype, "unlock", null);
    /**
     * @name Background Mode
     * @description
     * Cordova plugin to prevent the app from going to sleep while in background.
     * Requires Cordova plugin: cordova-plugin-background-mode. For more info about plugin, visit: https://github.com/katzer/cordova-plugin-background-mode
     * @usage
     * ```typescript
     * import { BackgroundMode } from '@ionic-native/background-mode';
     *
     * constructor(private backgroundMode: BackgroundMode) { }
     *
     * ...
     *
     * this.backgroundMode.enable();
     * ```
     *
     * @interfaces
     * BackgroundModeConfiguration
     */
    BackgroundMode = __decorate([
        Plugin({
            pluginName: 'BackgroundMode',
            plugin: 'cordova-plugin-background-mode',
            pluginRef: 'cordova.plugins.backgroundMode',
            repo: 'https://github.com/katzer/cordova-plugin-background-mode',
            platforms: ['AmazonFire OS', 'Android', 'Browser', 'iOS', 'Windows']
        })
    ], BackgroundMode);
    return BackgroundMode;
}(IonicNativePlugin));
export { BackgroundMode };
